policy/compressor.go

```diff
@@ -0,0 +1,127 @@
+package policy
+
+import (
+	"sync"
+
+	"go.uber.org/zap"
```

Comment:
Ahh, didn't realize we had our own logger in `m3x`. Do we want to use zap moving forward though? I don't feel strongly either way.